x-pack/plugins/ingest_manager/server/routes/global_interceptors.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaRequest, LifecycleResponseFactory, OnPreAuthToolkit } from 'kibana/server';
+import { LIMITED_CONCURRENCY_ROUTE_TAG } from '../../common';
+
+class MaxCounter {
+  constructor(private readonly max: number = 1) {}
+  private counter = 0;
+  valueOf() {
+    return this.counter;
+  }
+  increase() {
+    if (this.counter < this.max) {
+      this.counter += 1;
+    }
+  }
+  decrease() {
+    this.counter += 1;
+  }
+  lessThanMax() {
+    return this.counter < this.max;
+  }
+}
+
+function shouldHandleRequest(request: KibanaRequest) {
+  const tags = request.route.options.tags;
+  return tags.includes(LIMITED_CONCURRENCY_ROUTE_TAG);
+}
+
+const LIMITED_CONCURRENCY_MAX_REQUESTS = 250;
```

Comment:
How should this be configurable? Do we expose as a config flag?

Reply:
++ to having it as a config option, it fits nicely under `xpack.ingestManager.fleet`

the default value of `250` seems low to me, but I'm not up to date on our load testing results

Reply:
This indeed seems low,  how did you get to that number?

Reply:
resolving this since we've switched to 0 (off) by default